presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java

```diff
@@ -539,6 +556,8 @@ protected static class QueryContext
         private ImmutableList.Builder<String> setupQueryIds = ImmutableList.builder();
         private ImmutableList.Builder<String> teardownQueryIds = ImmutableList.builder();
 
+        private Optional<String> rewrittenFunctionCalls = Optional.empty();
```

Comment:
Is this a plural, should this be an Optional<List<String>> ?

Reply:
At this point, it's a comma concatenated strings of the rewritten function calls. Ex, `foo(a,b) is substituted by bar(a), baz(c) is substituted by qux(c)`.

I'm keeping a data member `private List<List<FunctionCall>> processedFunctionCallSubstitutes` in FunctionCallRewriter class to record what rewrites were done. Then `Optional<String> rewrittenFunctionCalls` here of AbstractVerification is set to `FunctionCallRewriter.getProcessedFunctionCallSubstitutes()` with implementation, 

```
    public String getProcessedFunctionCallSubstitutes()
    {
        return processedFunctionCallSubstitutes.stream().map(functionCallSubstitute -> {
                    String formattedOriginal = ExpressionFormatter.formatExpression(functionCallSubstitute.get(0), Optional.empty());
                    String formattedSubstitute = ExpressionFormatter.formatExpression(functionCallSubstitute.get(1), Optional.empty());
                    return String.format("%s is substituted with %s", formattedOriginal, formattedSubstitute);
                }
        ).collect(Collectors.joining(", "));
    }
```